components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAOTest.java

```diff
@@ -0,0 +1,1486 @@
+
+/**
+ * Unit tests for CacheBackedIdPManagementDAO.
+ */
+@PrepareForTest({IdentityDatabaseUtil.class, DataSource.class, IdentityTenantUtil.class, IdentityUtil.class,
+        CarbonContext.class})
+public class CacheBackedIdPMgtDAOTest extends PowerMockTestCase {
+
+    private static final String DB_NAME = "test";
+
+    private static final Integer SAMPLE_TENANT_ID = -1234;
+
+    private static final Integer SAMPLE_TENANT_ID2 = 1;
+
+    private static final Integer NOT_EXISTING_TENANT_ID = 4;
+
+    private static final String TENANT_DOMAIN = "carbon.super";
+
+    private static Map<String, BasicDataSource> dataSourceMap = new HashMap<>();
+
+    private CacheBackedIdPMgtDAO cacheBackedIdPMgtDAO;
+    private IdPManagementDAO idPManagementDAO;
+
+    private void initiateH2Database(String databaseName, String scriptPath) throws Exception {
+
+        BasicDataSource dataSource = new BasicDataSource();
+        dataSource.setDriverClassName("org.h2.Driver");
+        dataSource.setUsername("username");
+        dataSource.setPassword("password");
+        dataSource.setUrl("jdbc:h2:mem:test" + databaseName);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.createStatement().executeUpdate("RUNSCRIPT FROM '" + scriptPath + "'");
+        }
+        dataSourceMap.put(databaseName, dataSource);
+    }
+
+    private static String getFilePath(String fileName) {
+
+        if (StringUtils.isNotBlank(fileName)) {
+            return Paths.get(System.getProperty("user.dir"), "src", "test", "resources", "dbscripts", fileName)
+                    .toString();
+        }
+        throw new IllegalArgumentException("DB Script file name cannot be empty");
+    }
+
+    private static void closeH2Database() throws Exception {
+
+        BasicDataSource dataSource = dataSourceMap.get(DB_NAME);
+        if (dataSource != null) {
+            dataSource.close();
+        }
+    }
+
+    private static Connection getConnection(String database) throws SQLException {
+
+        if (dataSourceMap.get(database) != null) {
+            return dataSourceMap.get(database).getConnection();
+        }
+        throw new RuntimeException("No datasource initiated for database: " + database);
+    }
+
+    @BeforeMethod
+    public void setup() throws Exception {
+
+        idPManagementDAO = new IdPManagementDAO();
+        cacheBackedIdPMgtDAO = new CacheBackedIdPMgtDAO(idPManagementDAO);
+        initiateH2Database(DB_NAME, getFilePath("h2.sql"));
+        mockStatic(IdentityTenantUtil.class);
+        when(IdentityTenantUtil.getTenantDomain(anyInt())).thenReturn(TENANT_DOMAIN);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+
+        closeH2Database();
+    }
+
+    @DataProvider
+    public Object[][] getIdPsData() {
+
+        return new Object[][]{
+                {SAMPLE_TENANT_ID, TENANT_DOMAIN, 2},
+                {SAMPLE_TENANT_ID2, TENANT_DOMAIN, 1},
```

Comment:
All the case we are passing TENANT_DOMAIN there is not change in all 3 combination then no need to pass them right?

Reply:
Resolved with https://github.com/wso2/carbon-identity-framework/pull/4110/commits/28a1194ed9f5de659bd1879f625cd070efdac3fa